service/src/apps/area_name/area_name.go

```diff
@@ -0,0 +1,96 @@
+package area_name
+
+import (
+	"fmt"
+	"generic_apis/apps/utils"
+	"generic_apis/db"
+	"generic_apis/insight"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/microsoft/ApplicationInsights-Go/appinsights"
+)
+
+type handler struct {
+	db          *db.Config
+	traceparent string
+}
+
+func (conf *handler) fromDatabase(areaType, areaName string) (map[string]interface{}, error) {
+
+	var (
+		ok     bool
+		params []interface{}
+		query  string
+	)
+
+	areaType = strings.ToLower(areaType)
+
+	if areaType, ok = utils.AreaTypes[areaType]; !ok {
+		return nil, fmt.Errorf("invalid area type")
+	}
+
+	params = []interface{}{areaType, areaName}
+	query = areaQuery
+
+	payload := &db.Payload{
+		Query:         query,
+		Args:          params,
+		OperationData: insight.GetOperationData(conf.traceparent),
+	}
+	results, err := conf.db.FetchRow(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	return results, nil
+
+} // fromDatabase
+
+func Handler(insight appinsights.TelemetryClient) func(w http.ResponseWriter, r *http.Request) {
+
+	conf := &handler{}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		var (
+			err         error
+			response    map[string]interface{}
+			jsonPayload []byte
+		)
+
+		conf.traceparent = r.Header.Get("traceparent")
+
+		conf.db, err = db.Connect(insight)
+		if err != nil {
+			panic(err)
+		}
+
+		pathVars := mux.Vars(r)
+
+		response, err = conf.fromDatabase(pathVars["area_type"], pathVars["area_name"])
+		if err != nil {
+			http.Error(w, "failed to retrieve data", http.StatusBadRequest)
```

Comment:
Might be better to register this as `http.StatusInternalServerError`? Provides better telemetry in App Insight.